br/pkg/lightning/importer/import.go

```diff
@@ -630,7 +630,8 @@ func (worker *restoreSchemaWorker) addJob(sqlStr string, job *schemaJob) error {
 			zap.String("table", job.tblName),
 			zap.String("statement", sqlStr),
 			zap.Error(err))
-		job.stmts = []string{sqlStr}
+		useStmt := common.SprintfWithIdentifiers("USE %s", job.dbName)
+		job.stmts = []string{useStmt, sqlStr}
```

Comment:
we should return error if we cannot parse the sql, so user can see the real err. lightning version >= tidb-version in most case, so there shouldn't be cases that lightning cannot parse, but tidb can execute it.

Reply:
This is a cloud oncall, tiflow's version is lower than downstream TiDB

Reply:
then we should update tiflow with latest tidb dependency

also the logic to use original sql is error prone when the unparsable sql already contains dbname that might be the one we want

Reply:
Our rewrite logic will not change the error message if downstream can not execute them, the user still can see a meaningful error message which is reported by downstream TiDB's parser.

Reply:
i mean when the sql is unparsable by tiflow, but ok for tidb, then we might created it in the wrong db

Reply:
I agree `update tiflow with latest tidb dependency`

But since we already support the function of executing SQL even if it cannot be parsed, I think this PR can be fixed as a bug. Unless we firmly decide to remove it

Reply:
I should remove the fallback logic if tiflow can be newer than downstream TiDB

Reply:
I also agree with "update tiflow with latest tidb dependency"